e2e/test/scenarios/question/new.cy.spec.js

```diff
@@ -309,6 +310,67 @@ describe("scenarios > question > new", () => {
     });
     cy.get("header").findByText(NEW_COLLECTION);
   });
+
+  describe("add to a dashboard", () => {
+    const collectionInRoot = {
+      name: "Collection in root collection",
+    };
+    const dashboardInRoot = {
+      name: "Dashboard in root collection",
+    };
+    const myPersonalCollection = "My personal collection";
+
+    beforeEach(() => {
+      cy.createCollection(collectionInRoot);
+      cy.createDashboard(dashboardInRoot);
+      // Can't use `startNewQuestion` because it's missing `display: "table"` and
+      // adding that will fail a lot of other tests and I don't want to deal with that yet.
+      cy.visit("/");
+      cy.findByTestId("app-bar").button("New").click();
+      popover().findByText("Question").click();
+    });
+
+    it("should hide public collections when selecting a dashboard for a question in a personal collection", () => {
+      popover().within(() => {
+        cy.findByText("Raw Data").click();
+        cy.findByText("Orders").click();
+      });
+
+      queryBuilderHeader().button("Save").click();
+      modal().findByTestId("select-button").click();
+      popover().findByText("My personal collection").click();
+      modal().within(() => {
+        cy.button("Save").click();
+        cy.button("Yes please!").click();
```

Comment:
Here and in other places I would add `cy.wait` given with API calls, otherwise the test suite can be flaky

Reply:
Fixed in https://github.com/metabase/metabase/pull/35171/commits/158e4cf9bac0217b2636a05ce246cc2932a7835a